src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs

```diff
@@ -146,14 +150,42 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                     release.Seeders = ParseUtil.CoerceInt(row.QuerySelector("td:nth-of-type(7)").Text().Trim());
                     release.Peers = ParseUtil.CoerceInt(row.QuerySelector("td:nth-of-type(8)").Text().Trim()) + release.Seeders;
 
-                    var cat = row.QuerySelectorAll("td:nth-of-type(1) i").First().GetAttribute("class")
-                                            .Replace("torrent-icon", string.Empty)
-                                            .Replace("fa fa-", string.Empty)
-                                            .Replace("film", "1")
-                                            .Replace("tv", "2")
-                                            .Replace("music", "3")
-                                            .Replace("text-pink", string.Empty);
-                    release.Category = MapTrackerCatToNewznab(cat.Trim());
+                    var resolution = row.QuerySelector("span.badge-extra")?.TextContent.Trim();
+                    var catMatch = CatRegex.Match(row.QuerySelectorAll("td:nth-of-type(1) i").First().GetAttribute("class"));
+                    var cats = new List<int>();
+                    switch(catMatch.Groups[1].Value)
+                    {
+                        case "film":
+                            if (query.Categories.Contains(TorznabCatType.Movies.ID))
+                                cats.Add(TorznabCatType.Movies.ID);
+                            cats.Add(resolution switch
+                            {
+                                "2160p" => TorznabCatType.MoviesUHD.ID,
+                                "1080p" => TorznabCatType.MoviesHD.ID,
+                                "1080i" => TorznabCatType.MoviesHD.ID,
+                                "720p" => TorznabCatType.MoviesHD.ID,
+                                _ => TorznabCatType.MoviesSD.ID
+                            });
+                            break;
+                        case "tv":
+                            if (query.Categories.Contains(TorznabCatType.TV.ID))
+                                cats.Add(TorznabCatType.TV.ID);
+                            cats.Add(resolution switch
+                            {
+                                "2160p" => TorznabCatType.TVUHD.ID,
+                                "1080p" => TorznabCatType.TVHD.ID,
+                                "1080i" => TorznabCatType.TVHD.ID,
+                                "720p" => TorznabCatType.TVHD.ID,
+                                _ => TorznabCatType.TVSD.ID
+                            });
```

Comment:
According to https://stackoverflow.com/questions/56676260/c-sharp-8-switch-expression-multiple-cases-with-same-result the change below helps us avoid code duplication within the switch expressions.

```suggestion
                    var hdResolutions = new[] { "1080p", "1080i", "720p" };
                    switch(catMatch.Groups[1].Value)
                    {
                        case "film":
                            if (query.Categories.Contains(TorznabCatType.Movies.ID))
                                cats.Add(TorznabCatType.Movies.ID);
                            cats.Add(resolution switch
                            {
                                "2160p" => TorznabCatType.MoviesUHD.ID,
                                var res when hdResolutions.Contains(res) => TorznabCatType.MoviesHD.ID,
                                _ => TorznabCatType.MoviesSD.ID
                            });
                            break;
                        case "tv":
                            if (query.Categories.Contains(TorznabCatType.TV.ID))
                                cats.Add(TorznabCatType.TV.ID);
                            cats.Add(resolution switch
                            {
                                "2160p" => TorznabCatType.TVUHD.ID,
                                var res when hdResolutions.Contains(res) => TorznabCatType.TVHD.ID,
                                _ => TorznabCatType.TVSD.ID
                            });
```

Reply:
Changed.